### DISCUSSION TITLE
I18n and Typography not shown in generated template

### DISCUSSION THREAD

**Discussion by chmcquee** [keycloakify/keycloakify#249] [Q&A] [answered]
I recently tested out my theme building it with yarn keycloak and noticed none of my custom fonts, or i18n configurations were being injected as expected.

My file structure is such that all the relevant styling and typography as well as the i18n file is located in KcApp.
The File structure is:

src
i18n.ts
- components
    - styles
        - fonts.css
    - Login.tsx

- public
    - fonts
        - customFontsFolder

Anything I'm not aware of that needs fixing?

Thanks for your support.

**Comment by chmcquee** [comment]
So my colleague suggested changing the system language (mac) to the desired language. Now the i18n topic works as expected.  But the fonts are still not shown.

**Comment by garronej** [comment, marked as answer]
Hi @chmcquee,  

There is indeed a Keycloakify related limitation for fonts.  
There is an easy workaround though: Putting `@font-face` in the `index.html`.  

Read carfully this section https://docs.keycloakify.dev/limitations#font-face-importing-fonts-from-the-src-dir and you should be good

**Reply by chmcquee** [reply]
Excellent I must have missed this.
Thanks!

**Reply by chmcquee** [reply]
Hi @garronej,
while implementing this approach I have faced an issue:
Within the /public/fonts folder I also had a fonts.css file like you do in the demo project [here](https://github.com/garronej/keycloakify-demo-app/blob/9aa2dbaec28a7786d6b2983c9a59d393dec1b2d6/public/fonts/WorkSans/font.css) 
I was wondering if this file did anything?
Before I had to move my public file into the KcApp src file inorder to import the fonts.css to my App.tsx.
Looking through your code examples I never see an import of "fonts.css".
I was using my fonts.css for global styling of h1-h6.
Hope this makes sense and you can help me out.
Thanks again,
Chirs

**Reply by garronej** [reply]
> I was wondering if this file did anything?  

No it doesn't. It's not used nor imported anyhere, I should have deleted this file.  

You can import any .css files however you want as long as there is no `@fontface` in them.  

fontface must go in the index.html

**Reply by chmcquee** [reply]
Awesome thanks. This worked for me!
